src/draw/DefaultDrawer.js

```diff
@@ -106,466 +160,464 @@ class DefaultDrawer {
+        ?? this.shadowRoot;
+
+    const origIndex = origParent.children.findIndex((child) => child.id === event.subject.data.id);
+    const target = dropTarget ? d3.select(dropTarget) : null;
+
+    if (target) {
+      const parentId = target.attr('data-parentId');
+      const newParent = this.pluginData.getComponentById(parentId);
+
+      if (newParent.definition.childrenTypes.includes(event.subject.data.definition.type)) {
+        origParent.children.splice(origIndex, 1);
+        event.subject.data.setReferenceAttribute(newParent);
+        newParent.children.push(event.subject.data);
       }
+    } else {
+      origParent.children.splice(origIndex, 1);
+      event.subject.data.removeAllReferenceAttributes();
+      this.pluginData.components.push(event.subject.data);
     }
-    return {
-      sourceAnchor: 'top',
-      targetAnchor: 'top',
-    };
+    this.draw(this.rootId);
   }
 
   /**
-   * Set actions on viewport.
-   * @param {Object} element - D3 element related to the actions.
+   * Create a new svg to render the models in, or fetch an existing one.
    */
-  setViewPortAction(element) {
-    element.on('mousedown', () => {
-      this.__unselectComponent();
-      if (this.actions.link.targetSelector !== '') {
-        this.__unsetAddLink();
-      }
-    });
-  }
+  createRenderingContext() {
+    const contextIsPresent = !d3.select(`#${this.rootId}>svg`).empty();
 
-  /**
-   * Set selection actions on component.
-   * @param {Component} component - Component to set selection actions.
-   */
-  setSelectionAction(component) {
-    const element = this.d3.select(`#${component.id}`);
-    element.call(
-      this.d3.drag()
-        .on('start', () => {
-          this.actions.drag.state = false;
-        })
-        .on('drag', (_event, data) => {
-          this.__selectComponent(data.id);
-          this.actions.drag.state = true;
-          this.drawLinks();
-        })
-        .on('end', (_event, data) => {
-          if (!this.actions.drag.state) {
-            this.__toggleComponentSelection(data.id);
-          }
-          this.actions.drag.state = false;
-        }),
-    );
+    if (!contextIsPresent) {
+      this.svg = d3.select(`#${this.rootId}`)
+        .append('svg')
+        .attr('viewBox', [0, 0, this.width, this.height])
+        .attr('preserveAspectRatio', 'xMinYMin meet')
+        .style('font', '10px sans-serif');
+      this.svg.append('g')
+        .attr('class', 'container')
+        .attr('height', '100%')
+        .attr('width', '100%')
+        .attr('x', 0)
+        .attr('y', 0);
+    } else {
+      this.svg = d3.select(`#${this.rootId}`)
+        .select('svg');
+    }
   }
 
   /**
-   * Set actions on component.
-   * @param {Component[]} components - Array containing components.
+   * Draws all Components and ComponentLinks in the parentId Element.
+   * @param {String} [rootId] - Id of the container where you want to draw.
+   * @return {Boolean} Returns true if normally execute.
```

Comment:
Return is useless, mean remove return at the end of function

Reply:
done